src/main/java/com/hivemq/extensions/handler/PubackInterceptorHandler.java

```diff
@@ -0,0 +1,441 @@
+import com.hivemq.extensions.executor.PluginTaskExecutorService;
+import com.hivemq.extensions.executor.task.PluginInOutTask;
+import com.hivemq.extensions.executor.task.PluginInOutTaskContext;
+import com.hivemq.extensions.interceptor.puback.PubackInboundInputImpl;
+import com.hivemq.extensions.interceptor.puback.PubackInboundOutputImpl;
+import com.hivemq.extensions.interceptor.puback.PubackOutboundInputImpl;
+import com.hivemq.extensions.interceptor.puback.PubackOutboundOutputImpl;
+import com.hivemq.extensions.packets.puback.PubackPacketImpl;
+import com.hivemq.mqtt.message.puback.PUBACK;
+import com.hivemq.util.ChannelAttributes;
+import io.netty.channel.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * @author Yannick Weber
+ */
+@ChannelHandler.Sharable
+public class PubackInterceptorHandler extends ChannelDuplexHandler {
+
+    private static final Logger log = LoggerFactory.getLogger(PubackInterceptorHandler.class);
+
+    @NotNull
+    private final FullConfigurationService configurationService;
+
+    @NotNull
+    private final PluginOutPutAsyncer asyncer;
+
+    @NotNull
+    private final HiveMQExtensions hiveMQExtensions;
+
+    @NotNull
+    private final PluginTaskExecutorService pluginTaskExecutorService;
+
+    @Inject
+    public PubackInterceptorHandler(
+            @NotNull final FullConfigurationService configurationService,
+            @NotNull final PluginOutPutAsyncer asyncer,
+            @NotNull final HiveMQExtensions hiveMQExtensions,
+            @NotNull final PluginTaskExecutorService pluginTaskExecutorService) {
+        this.configurationService = configurationService;
+        this.asyncer = asyncer;
+        this.hiveMQExtensions = hiveMQExtensions;
+        this.pluginTaskExecutorService = pluginTaskExecutorService;
+    }
+
+    @Override
+    public void channelRead(final ChannelHandlerContext ctx, final Object msg) throws Exception {
+        if (!(msg instanceof PUBACK)) {
+            super.channelRead(ctx, msg);
+            return;
+        }
+        if (!handleInboundPuback(ctx, (PUBACK) msg)) {
+            super.channelRead(ctx, msg);
+        }
+    }
+
+    @Override
+    public void write(
+            @NotNull final ChannelHandlerContext ctx, @NotNull final Object msg, @NotNull final ChannelPromise promise)
+            throws Exception {
+        if (!(msg instanceof PUBACK)) {
+            super.write(ctx, msg, promise);
+            return;
+        }
+        if (!handleOutboundPuback(ctx, (PUBACK) msg, promise)) {
+            super.write(ctx, msg, promise);
+        }
+
+    }
+
+    private boolean handleOutboundPuback(
+            final @NotNull ChannelHandlerContext ctx,
+            final @NotNull PUBACK puback,
+            final @NotNull ChannelPromise promise) {
+        final Channel channel = ctx.channel();
+        if (!channel.isActive()) {
```

Comment:
Why is this check necessary?
The channel is always active in a channelRead event